$OpenBSD: patch-src_makemake_in,v 1.12 2015/10/06 11:37:06 jca Exp $

Override broken compiler test, attempting to build 32-bit clisp on a
64-bit patform will not work out very well.

Use the correct recipe to build shared modules on OpenBSD.

--- src/makemake.in.orig	Wed Jul  7 17:15:35 2010
+++ src/makemake.in     Sat Jun 10 13:57:52 2017
@@ -1150,6 +1150,7 @@ esac
 # esac
 
 test -z "$cpu" && cpu=${host_cpu}
+test "$cpu" = "${host_cpu%*64}" && cpu=${host_cpu}
 
 test "${verbose}" = true -o "${verbose}" = yes && \
   cat <<EOF >&2
@@ -1421,10 +1422,10 @@ if [ "${with_dynamic_modules}" != no ]; then
     freebsd2.2*)
       XCC_CREATESHARED='ld -Bshareable -o $lib $libs /usr/lib/c++rt0.o'
       ;;
-    freebsd2* | netbsd* | openbsd*)
+    freebsd2* | netbsd* )
       XCC_CREATESHARED='ld -Bshareable -o $lib $libs'
       ;;
-    freebsd3* | gnu* | linux* | cygwin* | mingw* | k*bsd* | dragonfly*)
+    freebsd3* | gnu* | linux* | cygwin* | mingw* | k*bsd* | dragonfly* | openbsd* )
       XCC_CREATESHARED='${CC} ${CFLAGS} ${CLFLAGS} -shared -o $lib $libs'
       ;;
     hpux9* | hpux10*)
@@ -1904,7 +1905,7 @@ echol "# This can also be used for pre-test and post-r
 echol "#   make distrib VERSION_SUFFIX=pre"
 echol "VERSION_SUFFIX="
 echol "# The distribution's top directory"
-echol 'TOPDIR=clisp-$(VERSION)$(VERSION_SUFFIX)'
+echol 'TOPDIR=clisp'
 echol
 echol '# Directories used by "make install":'
 echol "prefix = ${PREFIX}"
@@ -2931,7 +2932,7 @@ fi
 CLISP_="${HERE_}lisp${LEXE} -M ${HERE_}lispinit.mem ${localeflags}${someflags}"
 
 if [ $CROSS = false ] ; then
-  CHECK_DEPS="check-recompile check-fresh-line check-script check-exec-image check-tests"
+  CHECK_DEPS="check-recompile check-fresh-line check-script check-tests"
 
   echol "# check the sources:"
   echol "# 1. subr.d, fsubr.d, subrkw.d and all the LISPFUNs must add up"
@@ -3315,7 +3316,7 @@ for var in ${PRECIOUS_VARS}; do
 done
 # we must use $(SHELL) for sub-configures because when the top CONFIG_SHELL
 # is bash, config.cache may be unparsable with sh on Solaris
-echotab "      \$(SHELL) \$\$m/configure --with-clisp=\"${HEREP}/clisp -K boot ${someflags}\" --cache-file=\$\${cache} --srcdir=\$\$m \$(MODULE_CONFIGURE_FLAGS);\\"
+echotab "      \$(SHELL) \$\$m/configure --build=$(arch -s)-unknown-openbsd6.1 --with-clisp=\"${HEREP}/clisp -K boot ${someflags}\" --cache-file=\$\${cache} --srcdir=\$\$m \$(MODULE_CONFIGURE_FLAGS);\\"
 echotab "    else \\"
 echotab "      \$(SHELL) \$\$m/configure --srcdir=\$\$m \$(MODULE_CONFIGURE_FLAGS); \\"
 echotab "    fi ) ;\\"
